src/internal/m365/collection/exchange/collection.go

```diff
@@ -174,18 +176,46 @@ func NewCollection(
 	bc baseCollection,
 	user string,
 	items itemGetterSerializer,
+	added map[string]time.Time,
+	removed []string,
+	validModTimes bool,
 	statusUpdater support.StatusUpdater,
-) prefetchCollection {
-	collection := prefetchCollection{
+) data.BackupCollection {
+	add := maps.Clone(added)
+	remove := make(map[string]struct{}, len(removed))
+
+	// Remove any deleted IDs from the set of added IDs because items that are
+	// deleted and then restored will have a different ID than they did
+	// originally.
+	//
+	// TODO(ashmrtn): If we switch to immutable IDs then we'll need to handle this
+	// sort of operation in the pager since this would become order-dependent
+	// unless Graph started consolidating the changes into a single delta result.
```

Comment:
It is possible for an deletion to happen midway through us fetching pages from graph and thus we can never be certain that one item will not have multiple entries. I'm not sure how this plays into immutable IDs, but just wanted to mention it.

Reply:
I think we can catch duplicate IDs if changes happen mid-paging. The problem is if we have immutable IDs the handling of the duplicates becomes order dependent

For background, if an item is changed/moved/deleted while paging through results then either 1. the item will appear again in the result set or 2. the item will appear in the next delta result set in the next backup

The code below handles the first case by removing IDs in the added set if they also appear in the removed set. If we have multiple adds for the same item the map automatically consolidates them into a single item fetch

This would need to change if we switched to immutable IDs because in that case it would be possible to see a series of results like `(add item A, remove item A, add item A)`. The "correct" outcome in that case would be to fetch data for item A. Without immutable IDs the delta results look like `(add item A, remove item A, add item A')` so we avoid the ordering issue altogether